server/src/users/users.controller.ts

```diff
@@ -17,6 +27,29 @@ export class UsersController {
     return user;
   }
 
+  @Post('reset_password')
+  async resetPassword(@Request() req) {
+    //console.log(req.body.email)
+    let user;
+    try {
+      user = await this.usersService.findByEmail(req.body.email);
+    } catch (e) {
+      //console.error(e);
+    }
+    if (user) {
+      // console.log("got a user")
+      //console.log(user)
+      //great send email if you get here
+      //that email has a link to take you to update your password
+    } else {
+      // console.log("No User")
+      //do nothing
+    }
+    //reading online it looks like I might need to  make a table for reset-password.
+    //delete entries after password is reset OR if user is not found?
+    //this is basically to create and keep track of a token (that will expire)
+  }
```

Comment:
let's do something like
```
async resetPassword(
  @Request() req,
  @Response({ passthrough: true }) response: ResponseT,
) {
  try {
    const user = await this.usersService.findByEmail(req.body.email);
    // TODO send email to user
  } catch () {
    response.status(200).send();
  }
```

we shouldn't need to  make a table or delete any entries here
the email we send will have a link back to our page in it

"you clicked forgot your password. use <a href="nonprofitCircle.com/set-new-password?token=a_jwt_that_we_sign_with_a_secret_and_can_only_be_decoded_by_us">this link</a> to set your new password"
then, on that PATCH request with the new pw we'll update that user's row in the db

Reply:
should ResponseT be ResponseType ?

Reply:
nope. Never mind, I see it in auth.controller - it is ResponseT, I just wasn't importing it yet.